src/state/sagas/datasets/CreateDataset/CreateDataset.js

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+import { put, select, takeEvery, call } from 'redux-saga/effects';
+import { Api } from '../../../../services/config/Api';
+import { DATASET_ACTIONS_KEY } from '../../../commons/DatasetConstants';
+import { STATUSES } from '../../../commons/Constants';
+import { dispatchSetApplicationErrorMessage } from '../../../dispatchers/app/ApplicationDispatcher';
+import { t } from 'i18next';
+
+const getDatasets = (state) => state.dataset.list.data;
+
+export function* createDataset(action) {
+  const dataset = action.dataset;
+  const organizationId = action.organizationId;
+  try {
+    const datasets = [...(yield select(getDatasets))];
```

Comment:
Copying the whole list of datasets may lead to performance issues when we have 15k+ datasets. Maybe we could use the action `ADD_DATASET` to simply add the new dataset to the list?

Reply:
good idea, thanks ! :pray: